chalice/deploy/deployer.py

```diff
@@ -296,22 +297,22 @@ def __init__(self, aws_client):
         # type: (TypedAWSClient) -> None
         self._aws_client = aws_client
 
-    def deploy(self, config, existing_resources):
-        # type: (Config, Optional[DeployedResources]) -> Tuple[str, str, str]
+    def deploy(self, config, existing_resources, lambda_arn):
+        # type: (Config, OPT_RESOURCES, str) -> Tuple[str, str, str]
         if existing_resources is not None and \
                 self._aws_client.rest_api_exists(
                     existing_resources.rest_api_id):
             print "API Gateway rest API already found."
             rest_api_id = existing_resources.rest_api_id
-            return self._create_resources_for_api(config, rest_api_id)
+            return self._create_resources_for_api(config, rest_api_id,
+                                                  lambda_arn)
         else:
             print "Initiating first time deployment..."
-            return self._first_time_deploy(config)
+            return self._first_time_deploy(config, lambda_arn)
```

Comment:
It seems like you flip-flop between calling it `handler_arn` and `lambda_arn`. Given that this PR is titled remove references to `lambda_arn`, it seems like you should stick with `handler_arn` to make it clearer.